expat/CMakeLists.txt

```diff
@@ -270,3 +271,34 @@ if(BUILD_tests)
     target_link_libraries(runtestspp expat)
     add_test(runtestspp tests/runtestspp)
 endif(BUILD_tests)
+
+if(BUILD_fuzz)
+    if(NOT "${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+        message(FATAL_ERROR
+            "Building fuzz targets without clang is not supported. Please set "
+            "-DCMAKE_C_COMPILER=clang.")
+    endif()
+
+    string(FIND "${CMAKE_EXE_LINKER_FLAGS}" "-fsanitize" sanitizer_present)
+    if(${sanitizer_present} EQUAL "-1")
+        message(WARNING
+            "There was no sanitizer present when building the fuzz targets. "
+            "This is likely in error - consider adding "
+            "-DCMAKE_C_FLAGS='-fsanitize=<sanitizer>' and "
+            "-DCMAKE_EXE_LINKER_FLAGS='-fsanitize=<sanitizer>' to your cmake "
+            "execution.")
+    endif()
```

Comment:
This is not for fuzz-coverage generation, this is a hint that you should probably also be fuzzing with ASan/MSan/UBSan :)

Reply:
I see — nice idea. Can we add that "ASan/MSan/UBSan" help into "sanitizer present" in the help text as well? I think it would explain itself a lot more.